initial setup

aaqibqadeer opened the pull request.

aaqibqadeer committed:
initial setup, basic react functionality

aaqibqadeer committed:
initial setup

Lutif commented on packages/shopping-cart/src/pages/Login.jsx:
```diff
@@ -0,0 +1,22 @@
+    <div className="bg-light p-5 my-5 col-5 mx-auto border border-2">
+      <h3 className="text-center mb-4">Enter your credentials</h3>
+      <form className="">
+        <div className="mb-3">
```
extract similar items into reusable components such as inputs , labels for forms

Lutif commented on packages/shopping-cart/src/pages/Login.jsx:
```diff
@@ -0,0 +1,22 @@
+import {Link} from 'react-router-dom';
+export default function Login(props) {
+  return(
```
where is state logic ? should be storing data in state.

Lutif commented on packages/shopping-cart/src/pages/Product.js:
```diff
@@ -0,0 +1,14 @@
+export default function Product(props) {
+  let product = props.product
```
is there reassignment of product variable?

Lutif commented on packages/shopping-cart/src/pages/Product.js:
```diff
@@ -0,0 +1,14 @@
+export default function Product(props) {
+  let product = props.product
+  return(
+    <div className="card col-3" style={{width: "18rem"}}>
```
avoid inline styles

Lutif replied:
avoid inline styles

Lutif commented on packages/shopping-cart/src/pages/Register.jsx:
```diff
@@ -0,0 +1,30 @@
+        <div className="mb-3">
+          <label className="form-label">Name:</label>
+          <input type="text" className="form-control" placeholder="Enter your name" />
+        </div>
```
repeated jsx extract to a reusable component

Lutif commented on packages/shopping-cart/src/pages/Register.jsx:
```diff
@@ -0,0 +1,30 @@
+import {Link} from 'react-router-dom';
+export default function Register(props) {
```
add a line space for line:2

Lutif commented on packages/shopping-cart/src/pages/Register.jsx:
```diff
@@ -0,0 +1,30 @@
+import {Link} from 'react-router-dom';
+export default function Register(props) {
+  return(
+    <div className="bg-light p-5 my-5 col-5 mx-auto border border-2">
```
add a line space before return

Lutif commented on packages/shopping-cart/src/pages/Products.js:
```diff
@@ -0,0 +1,15 @@
+import productsList from '../products.json';
+import Product from './Product';
```
avoid default imports, prefer named imports

Lutif replied:
^2

Lutif commented on packages/shopping-cart/src/routes/routes.js:
```diff
@@ -0,0 +1,10 @@
+import Home from "./pages/Home";
+import Products from "./pages/Products";
+
+
```
extra linespace

Lutif replied:
^2

Lutif commented on packages/shopping-cart/src/App.tsx:
```diff
@@ -0,0 +1,50 @@
+  return (
+    <div className="App">
+      <Router>
+        <nav className="mb-5 navbar navbar-dark bg-dark navbar-expand-lg">
```
extract navbar

aaqibqadeer committed:
basic configuration

aaqibqadeer committed:
- extract components for reusability
- add line spaces

aaqibqadeer committed:
add login component logic

aaqibqadeer committed:
Add register component logic

aaqibqadeer committed:
add spinner option

aaqibqadeer committed:
add form validation conditions

aaqibqadeer committed:
add cart page

Lutif commented on packages/shopping-cart/src/components/InputField.jsx:
```diff
@@ -0,0 +1,19 @@
+  );
+}
+
+export default InputField;
```
avoid default exports - pointed out in earlier review as well

Lutif commented on packages/shopping-cart/src/pages/Carousel.jsx:
```diff
@@ -0,0 +1,42 @@
+      </div>
+      <div className="carousel-inner">
+        <div className="carousel-item active" data-bs-interval="10000">
+          <img src="..." className="d-block w-100" alt="..."/>
```
extract repeated JSX to the reusable component

Lutif commented on packages/shopping-cart/src/pages/Header.jsx:
```diff
@@ -0,0 +1,53 @@
+            { authStatus && <button onClick={logout} className="btn btn-primary mx-2">Sign out</button> }
+            { !authStatus && <Link to="/login" className="btn btn-primary mx-2">Sign In</Link> }
+        </nav>
+        <Switch>
```
wrong place to have routing logic

Lutif commented on packages/shopping-cart/src/pages/Login.jsx:
```diff
@@ -0,0 +1,57 @@
+  function handleSubmit(event) {
+    event.preventDefault();
+    if(userExists()) {
+      setTimeout(() => {
```
always clear time out ids when done with it.

Lutif commented on packages/shopping-cart/src/pages/Register.jsx:
```diff
@@ -0,0 +1,55 @@
+    }
+  }
+
+  
```
extra line

aaqibqadeer committed:
convert default imports to named imports

aaqibqadeer committed:
Refactor header and router logic
- Extracted nav, links and signout component
- Move router logic to router component
- Rendering routes and links through:xqops

aaqibqadeer committed:
Removed usused variables

aaqibqadeer committed:
clearing setTimeout in everywhere it is being used

aaqibqadeer committed:
Restore route logic into app component

aaqibqadeer committed:
Code restructure

aaqibqadeer committed:
extract out loop from jsx

aaqibqadeer committed:
indent product component

aaqibqadeer committed:
Add display message for no products

aaqibqadeer committed:
use useEffect hook inside cart page

aaqibqadeer committed:
Refactor styling

- convert inline styles to external cs
- add styled component

aaqibqadeer committed:
fix small typo

aaqibqadeer committed:
add navigation guard

aaqibqadeer committed:
merge

aaqibqadeer committed:
bug fixes

aaqibqadeer committed:
fixes css issues

aaqibqadeer committed:
fixe a small issue on product page

aaqibqadeer committed:
fix small style issues

aaqibqadeer committed:
fixed subtotal price issue
